apps/s_client.c

```diff
@@ -2190,6 +2193,25 @@ int s_client_main(int argc, char **argv)
                 if (strstr(mbuf, "STARTTLS"))
                     foundit = 1;
             } while (mbuf_len > 1 && mbuf[0] != '.');
+    case PROTO_SIEVE:
+        {
+            int foundit = 0;
+            BIO *fbio = BIO_new(BIO_f_buffer());
+
+            BIO_push(fbio, sbio);
+            /* wait for multi-line response to end from Sieve */
+            do {
+                mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+                /*
+                 * According to RFC 5804 § 1.7, capability
+                 * is case-insensitive, make it uppercase
+                 */
+                if (mbuf_len > 1 && mbuf[0] == '"') {
+                    make_uppercase(mbuf);
+                    if (strstr(mbuf, "\"STARTTLS\"") == mbuf)
```

Comment:
does it make sense to put that do/while loop into common code for nntp to also use?